### DISCUSSION TITLE
brew on NFS: error Another active Homebrew vendor-install-ruby process is already in progress

### DISCUSSION THREAD

**Discussion by nikonakoneko** [Homebrew/discussions#3334] [Getting started] [answered]
I'm trying to install `brew` on a NFS volume. `mount` gives me the following options for the mount point: `nfs, asynchronous, nodev, nosuid, automounted, noowners, noatime, nobrowse`.

Once the brew repository cloned I can run `brew shellenv`, but any other command will fail with the following error:
```
Traceback (most recent call last):
	1: from -e:1:in `<main>'
-e:1:in `flock': Operation not supported (Errno::ENOTSUP)
Error: Another active Homebrew vendor-install-ruby process is already in progress.
Please wait for it to finish or terminate it to continue.
Error: Failed to install Homebrew Portable Ruby (and your system version is too old)!
```

Steps to reproduce:
```bash
# From inside the NFS mounted volume
git clone --depth 1 https://github.com/Homebrew/brew brew_test
eval "$(./brew_test/bin/brew shellenv)"
brew update --force
```

On local drive it runs fine. But I have not enough storage. (and no root access)

**Comment by Bo98** [comment, marked as answer]
Looks like your NFS server does not support file locking. We perhaps should handle this scenario better, though we've equally not had anybody complain about it.

It should however be possible to enable file locking under NFS via `lockd`. How you do that will depend on your NFS server setup.

**Reply by nikonakoneko** [reply]
Well, without super user powers... anyway thanks a lot for the explaination! ;)
